### PR TITLE
APPS-634 App Template: add new npm script

### PR DESCRIPTION
**Pull request by deivamagalhaes**
[Jira issue](https://helpscout.atlassian.net/browse/APPS-634)

### Problem

Because of the changes done in [HSAPP-8506](https://helpscout.atlassian.net/browse/HSAPP-8506), we can’t use npm run dev to start our local Vite apps anymore. We need the app to be built (for production) and exposed through a network IP.

This can be achieved by running `npm run build && npm run preview -- --host`, but we don't want developers to have to know that...

### Solution

Add a new npm script to watch for code changes, build and start the app with the --host param.

I've also update the other scripts to include the `--host` param, just in case someone uses them. 😬 

### Testing

1. Run the script by executing `npm run watch`
    - [x] You can see in the output that the app is built and a network host is provided
2. Change the main app component somehow
    - [x] You can see in the output that the app is rebuilt

[HSAPP-8506]: https://helpscout.atlassian.net/browse/HSAPP-8506?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by deivamagalhaes** [comment]
@Charca , do we need to change anything for Code Sandbox or will it magically work there? How does it know which script to run? I could not find it...

File: package.json [hunk 1]
```diff
@@ -4,9 +4,10 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
-    "dev": "vite",
+    "dev": "vite --host",
     "build": "tsc && vite build",
-    "preview": "vite preview"
+    "preview": "vite preview --host",
+    "watch": "tsc && vite preview --host & && vite build --watch"
```
Comment (tamarazuk) [inline package.json]:
```suggestion
    "watch": "tsc && vite preview --host && vite build --watch"
```

**Review by tamarazuk — CHANGES_REQUESTED** [review]
Looks good, just ran into a small typo

File: package.json (hunk 1 above)
Comment (deivamagalhaes) [reply package.json]: Not a typo. 😬 
It's so that the preview runs in the background and we can execute the next command.

Comment (tamarazuk) [reply package.json]: oh okay, it was causing this error on my end 😓

Comment (tamarazuk) [reply package.json]:
```
tamarazuk@Tamaras-MacBook-Pro app-template % npm run watch

> codesandbox-template-vite@0.0.0 watch
> tsc && vite preview --host & && vite build --watch

sh: -c: line 0: syntax error near unexpected token `&&'
sh: -c: line 0: `tsc && vite preview --host & && vite build --watch'
```

**Review by tamarazuk — APPROVED** [review]
thank you for bringing back the parens :)

File: package.json (hunk 1 above)
Comment (deivamagalhaes) [reply package.json]: Removing the parenthesis was the "typo". 😬

Comment (deivamagalhaes) [reply package.json]: It worked when I tested it, but it was probably cached. 🤦‍♀️

**Pull request merged**